File: launcher/src/main/scala/launcher/config/Movementconfig.scala
package launcher.config

import domain.ports.persistence.MovementPersistencePort
import domain.ports.service.MovementServicePort
import domain.service.MovementServiceImpl
import org.springframework.context.annotation.{Bean, Configuration}

@Configuration class MovementConfig {
  @Bean def movementPersistencePort(
      movementPersistencePort: MovementPersistencePort
  ) = new MovementServiceImpl(movementPersistencePort)
}

File: launcher/src/main/java/launcher/config/MovementConfig.java
package launcher.config;

import domain.ports.persistence.MovementPersistencePort;
import domain.ports.service.MovementServicePort;
import domain.service.MovementServiceImpl;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class MovementConfig {

    @Bean
    public MovementServicePort movementPersistencePort(final MovementPersistencePort movementPersistencePort) {
        return new MovementServiceImpl(movementPersistencePort);
    }

}
